cmd/kusk/cmd/deploy.go

```diff
@@ -0,0 +1,197 @@
+/*
+The MIT License (MIT)
+
+Copyright © 2022 Kubeshop
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/kubeshop/kusk-gateway/api/v1alpha1"
+	fileWatcher "github.com/kubeshop/kusk-gateway/cmd/kusk/internal/mocking/filewatcher"
+	"github.com/kubeshop/kusk-gateway/cmd/kusk/internal/utils"
+	"github.com/kubeshop/kusk-gateway/cmd/kusk/templates"
+	"github.com/kubeshop/kusk-gateway/pkg/options"
+	"github.com/kubeshop/kusk-gateway/pkg/spec"
+	"github.com/kubeshop/testkube/pkg/ui"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	file  string
+	watch bool
+)
+
+func init() {
+	//add to root command
+	rootCmd.AddCommand(deployCmd)
+
+	deployCmd.Flags().StringVarP(&file, "file", "f", "", "file to deploy")
+	deployCmd.MarkFlagRequired("file")
+
+	deployCmd.Flags().BoolVarP(&watch, "watch", "w", false, "watch file changes and deploy on change")
+	deployCmd.Flags().StringVar(&name, "name", "", "name to name API with ")
+
+}
+
+// apiCmd represents the api command
+var deployCmd = &cobra.Command{
+	Use:   "deploy",
+	Short: "deploy command to deploy your apis",
+	Long:  ``,
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := context.Background()
+		cmd.SilenceUsage = true
+		originalManifest, err := getParsedAndValidatedOpenAPISpec(file)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+		k8sclient, err := utils.GetK8sClient()
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+
+		api := &v1alpha1.API{}
+		yaml.Unmarshal([]byte(originalManifest), api)
+		if len(api.Namespace) == 0 {
+			api.Namespace = envoyFleetNamespace
+		}
+
+		if err := k8sclient.Create(ctx, api, &client.CreateOptions{}); err != nil {
+			fmt.Fprintln(os.Stderr, err)
```

Comment:
Up to you, if you want to change it, but there is also `RunE`, please see: https://github.com/spf13/cobra/blob/main/user_guide.md#returning-and-handling-errors.